lib/queues.js

```diff
@@ -611,5 +605,53 @@ module.exports = class Queues {
     // Lazy load configuration.
     return this._ironium.configPromise;
   }
-};
 
+  // Resolves to the maximum amount of jobs a queue should try to reserve.
+  // Promise doesn't resolve until capacity is available.
+  getReserveCapacity() {
+    return this._calculateReserveCapacity()
+      .then(capacity => {
+        if (capacity > 0)
+          return capacity;
+        else {
+          return new Promise(resolve => {
+            this._waitingForReserve.push(resolve);
+          });
+        }
+      });
+  }
+
+  incrementActiveJobs(count) {
```

Comment:
Algorithm looks solid, this function though needs a better name. It doesn't increment active jobs by count, rather allocates capacity for up to that many jobs that should run next.